src/sentry/static/sentry/app/views/performance/transactionSummary/keyTransactionButton.tsx

```diff
@@ -0,0 +1,154 @@
+  keyFetchID: symbol | undefined;
+  error: null | string;
+
+  isKeyTransaction: boolean;
+};
+
+class KeyTransactionButton extends React.Component<Props, State> {
+  state: State = {
+    isLoading: true,
+    keyFetchID: undefined,
+    error: null,
+
+    isKeyTransaction: false,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const orgSlugChanged = prevProps.organization.slug !== this.props.organization.slug;
+    const projectsChanged =
+      prevProps.eventView.project.length === 1 &&
+      this.props.eventView.project.length === 1 &&
+      prevProps.eventView.project[0] !== this.props.eventView.project[0];
+
+    if (orgSlugChanged || projectsChanged) {
+      this.fetchData();
+    }
+  }
+
+  fetchData = () => {
+    const {organization, eventView, transactionName} = this.props;
+
+    const projects = eventView.project as number[];
+
+    if (projects.length !== 1) {
+      return;
+    }
+
+    const url = `/organizations/${organization.slug}/is-key-transactions/`;
+    const keyFetchID = Symbol('keyFetchID');
+
+    this.setState({isLoading: true, keyFetchID});
+
+    this.props.api
+      .requestPromise(url, {
+        method: 'GET',
+        includeAllArgs: true,
+        query: {
+          project: projects.map(id => String(id)),
+          transaction: transactionName,
+        },
+      })
+      .then(([data, _, _jqXHR]) => {
+        if (this.state.keyFetchID !== keyFetchID) {
+          // invariant: a different request was initiated after this request
+          return;
+        }
+
+        this.setState({
+          isLoading: false,
+          keyFetchID: undefined,
+          error: null,
+          isKeyTransaction: !!data?.isKey,
+        });
+      })
+      .catch(err => {
+        this.setState({
+          isLoading: false,
+          keyFetchID: undefined,
+          error: err.responseJSON.detail,
+          isKeyTransaction: false,
+        });
+      });
+  };
+
+  toggleKeyTransaction = () => {
+    const {eventView, api, organization, transactionName} = this.props;
+    const projects = eventView.project as number[];
+
+    if (!this.state.isKeyTransaction) {
+      this.setState({
+        isKeyTransaction: true,
+      });
+
+      saveKeyTransaction(api, organization.slug, projects, transactionName).catch(() => {
+        this.setState({
+          isKeyTransaction: false,
+        });
+      });
+    } else {
+      this.setState({
+        isKeyTransaction: false,
+      });
+
+      deleteKeyTransaction(api, organization.slug, projects, transactionName).catch(
+        () => {
+          this.setState({
+            isKeyTransaction: true,
+          });
+        }
+      );
+    }
+  };
+
+  render() {
+    const {isKeyTransaction, isLoading} = this.state;
+
+    if (this.props.eventView.project.length !== 1 || isLoading) {
+      return null;
+    }
+
+    return (
+      <Button onClick={this.toggleKeyTransaction}>
```

Comment:
Button components have an `icon` prop that takes an icon component. Using that prop would let you skip styling the button to add a margin.